src/Symfony/Bundle/WebProfilerBundle/Profiler/CodeExtension.php

```diff
@@ -25,15 +25,13 @@
  */
 final class CodeExtension extends AbstractExtension
 {
-    private string|FileLinkFormatter|array|false $fileLinkFormat;
-    private string $charset;
-    private string $projectDir;
-
-    public function __construct(string|FileLinkFormatter $fileLinkFormat, string $projectDir, string $charset)
-    {
+    public function __construct(
+        private string|FileLinkFormatter|array|false $fileLinkFormat,
```

Comment:
I don't think we want to widen the constructor parameter type this way. Rule of thumb: if the parameter type and the property type don't match, we don't promote.

Reply:
Ok sorry, i revert

Reply:
✅

Reply:
(thank you for the review 😄)